provider/adapters/src/fork/by_error.rs

```diff
@@ -142,10 +142,20 @@ impl<P, F> MultiForkByErrorProvider<P, F> {
         &self.providers
     }
 
+    /// Returns a mutable reference to the inner vector of providers.
+    pub fn inner_mut(&mut self) -> &mut Vec<P> {
+        &mut self.providers
+    }
+
     /// Returns ownership of the inner providers to the caller.
     pub fn into_inner(self) -> Vec<P> {
         self.providers
     }
+
+    /// Adds an additional child provider.
+    pub fn push(&mut self, provider: P) {
```

Comment:
This can be done as `.inner_mut().push(...)`, but I think it will be the most common operation, so I made it its own function

Reply:
Also if we _do_ change the API in the future over a 2.0 boundary, most clients will still work.